integrations/hermes-agent/tests/tools/test_notte_provider.py

```
@@ -0,0 +1,358 @@
+
+        debug_response = Mock()
+        debug_response.ok = True
+        debug_response.json.return_value = {"debug_url": "https://debug.notte.cc"}  # missing ws.cdp
+        debug_response.text = '{"debug_url": "https://debug.notte.cc"}'
+
+        with patch("tools.browser_providers.notte.requests") as mock_requests:
+            mock_requests.post.return_value = start_response
+            mock_requests.get.return_value = debug_response
+            mock_requests.delete.return_value = Mock(status_code=200)
+
+            with pytest.raises(RuntimeError, match="missing 'ws.cdp'"):
```

Ruff (0.15.9)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)